include/clang/Sema/DeclSpec.h

```diff
@@ -393,8 +393,8 @@ class DeclSpec {
 
   TypedefDecl **TypeVarInfo;
   unsigned NumTypeVars : 15;
-  bool GenericFunction : 1;
-  bool ItypeGenericFunction : 1;
+  bool GenericFunctionOrStruct : 1;
```

Comment:
Is there a reason to combine the bit fields for generic function and struct and not keep them separate for more granularity? Similarly, for the methods below.

Reply:
We can reuse the current logic in DeclSpec with minimal changes. If we need to disambiguate between them in the future, I propose we split them at that point.